app/controllers/shelters_controller.rb

```diff
@@ -11,12 +11,18 @@ def new
   end
 
   def create
-    shelter = Shelter.create(name: params[:shelter_name],
+    shelter = Shelter.new(name: params[:shelter_name],
               address: params[:shelter_address],
               city: params[:shelter_city],
               state: params[:shelter_state],
               zip: params[:shelter_zip])
-    redirect_to '/shelters'
+    if shelter.save
+      redirect_to '/shelters'
+      flash[:success] = "The shelter has been created!"
+    else
+      flash.now[:notice] = shelter.errors.full_messages.to_sentence
+      render :new
+    end
```

Comment:
Changed this to .new so that we could add a conditional IF the shelter is created. Since I am now validating presence of all items, the shelter will not create without all fields filled out and if any aren't filled out, they will get a message telling them which field cannot be empty.

Reply:
Perfect, I will be using the same process to validate fields in pet and flashing this exact error message.